### PR TITLE
renaming fork-previous-global-slot to fork-genesis-slot

### LINKED ISSUE #13897: Global slot of link block at hard fork is the previous slot
When providing hard fork parameters, including the previous global slot, a block is created without transactions whose parent state hash is the previous state hash given in the parameters.

The global slot since genesis for this "link block" is the previous global slot in the hard fork parameters.

Is that the correct slot?

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
Explain your changes:
Renaming `fork_previous_global_slot` to `fork_genesis_global_slot`. We are doing this change because when generating the new genesis block for a hard-forked chain, we are taking the value of `fork_previous_global_slot` as the new genesis slot number. So if we give it the `slot` number of previous block, then we could have 2 blocks that have same slot number on the chain. This renaming is intended to inform the user that this `slot` number is for the new genesis block, so it should be given a value such as `previous_global_slot + 1`.

Explain how you tested your changes:
*

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #13897

### CODE REVIEW

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

File: src/lib/genesis_constants/genesis_constants.ml [hunk 1]
```diff
@@ -174,8 +174,7 @@ module Constraint_constants = struct
                 fork_previous_state_hash
           ; previous_length = Mina_numbers.Length.of_int fork_previous_length
           ; previous_global_slot =
```
Comment (psteckler) [inline src/lib/genesis_constants/genesis_constants.ml:176]: does this record field need to change, also?

**Review by psteckler — COMMENTED** [review]
I think you'll need to change the field name in type `Fork_constants.t` and also `Runtime_config.Fork_config.t`.

That affects the hard fork integration test, which isn't run in PR CI.

**Review by psteckler — APPROVED** [review]

**Comment by psteckler** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Review by mitschabaude — APPROVED** [review]

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Review by deepthiskumar — APPROVED** [review]

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**